modules/ble/src/ble_instr_utils.c

```diff
@@ -203,53 +203,85 @@ VECTOR_HANDLE parse_instructions(JSON_Array* instructions)
                     }
                     else
                     {
-                        const char* characteristic_uuid = json_object_get_string(instr, "characteristic_uuid");
-                        if (characteristic_uuid == NULL)
-                        {
-                            /*Codes_SRS_BLE_05_009: [ BLE_CreateFromJson shall return NULL if a given instruction does not have a characteristic_uuid property. ]*/
-                            LogError("json_object_get_string returned NULL for the property 'characteristic_uuid' for instruction number %zu", i);
-                            free_instructions(result);
-                            VECTOR_destroy(result);
-                            result = NULL;
-                            break;
-                        }
-                        else
-                        {
-                            BLE_INSTRUCTION ble_instr = { 0 };
-                            ble_instr.characteristic_uuid = STRING_construct(characteristic_uuid);
-                            if (ble_instr.characteristic_uuid == NULL)
+                        if ( strcmp(type, "sequential") == 0) {
+                            JSON_Array* seq_instructions = json_object_get_array(instr, "instructions");
+                            VECTOR_HANDLE seqInsts = parse_instructions(seq_instructions);
+                            if (seqInsts!=NULL){
+                                size_t numOfSeqs = VECTOR_size(seqInsts);
+                                if (numOfSeqs>0){
+  									BLE_INSTRUCTION* prev_instr = (BLE_INSTRUCTION*)VECTOR_element(seqInsts, 0);
+									BLE_INSTRUCTION firstInstr = { 0 };
+									firstInstr.instruction_type = prev_instr->instruction_type;
+									firstInstr.characteristic_uuid = STRING_clone(prev_instr->characteristic_uuid);
+									firstInstr.nextInst = NULL;
+									memcpy(&(firstInstr.data), &(prev_instr->data), sizeof(prev_instr->data));
+									prev_instr = &firstInstr;
+									if (numOfSeqs > 1) {
+										for (size_t c = 1; c < numOfSeqs; c++) {
+											BLE_INSTRUCTION* current = (BLE_INSTRUCTION*)VECTOR_element(seqInsts, c);
+											BLE_INSTRUCTION* currentInstr = (BLE_INSTRUCTION*)malloc(sizeof(BLE_INSTRUCTION));
+											currentInstr->instruction_type = current->instruction_type;
+											currentInstr->characteristic_uuid = STRING_clone(current->characteristic_uuid);
+											memcpy(&(currentInstr->data), &(current->data), sizeof(current->data));
+											prev_instr->nextInst = currentInstr;
+											prev_instr = currentInstr;
+										}
+									}
+									if (VECTOR_push_back(result, &firstInstr, 1) != 0) {
+										// error
+									}
```

Comment:
Overall, I like what you're doing here, but there's almost no error-checking. 
If the parse_instructions() call returns NULL, this function should clean up and return NULL, same for the inability to malloc, clone and pushing onto the vector.